pkg/downloader/downloader.go

```diff
@@ -274,9 +291,43 @@ func copyLocal(dst, src string, description string, expectedDigest digest.Digest
 	if description != "" {
 		// TODO: progress bar for copy
 	}
+	if IsCompressed(ext) && decompress {
+		return decompressLocal(dstPath, srcPath, ext)
+	}
 	return fs.CopyFile(dstPath, srcPath)
 }
 
+func decompressLocal(dst, src, ext string) error {
+	var program string
+	switch ext {
+	case ".gz":
+		program = "gzip"
+	case ".bz2":
+		program = "bzip2"
+	case ".xz":
+		program = "xz"
+	case ".zst":
+		program = "zstd"
+	default:
+		return fmt.Errorf("decompressLocal: unknown extension %s", ext)
+	}
+	logrus.Infof("decompressing %s with %s", ext, program)
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.OpenFile(dst, os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	cmd := exec.Command(program, "-d") // --decompress
+	cmd.Stdin = in
+	cmd.Stdout = out
+	return cmd.Run()
```

Comment:
Can we return stderr string as the err on an error?

Reply:
Done, returning ExitError